exporter/lokiexporter/exporter.go

```diff
@@ -227,14 +229,68 @@ func (l *lokiExporter) convertAttributesToLabels(attributes pdata.AttributeMap,
 	return ls
 }
 
-func convertLogBodyToEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
+func (l *lokiExporter) convertLogBodyToEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
+	var b strings.Builder
+
+	if len(lr.Name()) > 0 {
+		b.WriteString("name=")
+		b.WriteString(lr.Name())
+		b.WriteRune(' ')
+	}
+	if len(lr.SeverityText()) > 0 {
+		b.WriteString("severity=")
+		b.WriteString(lr.SeverityText())
+		b.WriteRune(' ')
+	}
+	if lr.SeverityNumber() > 0 {
+		b.WriteString("severityN=")
+		b.WriteString(strconv.Itoa(int(lr.SeverityNumber())))
+		b.WriteRune(' ')
+	}
+	if !lr.TraceID().IsEmpty() {
+		b.WriteString("traceID=")
+		b.WriteString(lr.TraceID().HexString())
+		b.WriteRune(' ')
+	}
+	if !lr.SpanID().IsEmpty() {
+		b.WriteString("spanID=")
+		b.WriteString(lr.SpanID().HexString())
+		b.WriteRune(' ')
+	}
+
+	// fields not added to the accept-list as part of the component's config
+	// are added to the body, so that they can still be seen under "detected fields"
+	lr.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+		if _, found := l.config.Labels.Attributes[k]; !found {
+			b.WriteString(k)
+			b.WriteString("=")
+			b.WriteString(v.AsString())
+			b.WriteRune(' ')
+		}
+		return true
+	})
+
+	// same for resources: include all, except the ones that are explicitly added
+	// as part of the config, which are showing up at the top-level already
+	res.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+		if _, found := l.config.Labels.ResourceAttributes[k]; !found {
+			b.WriteString(k)
+			b.WriteString("=")
+			b.WriteString(v.AsString())
+			b.WriteRune(' ')
+		}
+		return true
+	})
+
+	b.WriteString(lr.Body().StringVal())
+
 	return &logproto.Entry{
 		Timestamp: time.Unix(0, int64(lr.Timestamp())),
-		Line:      lr.Body().StringVal(),
+		Line:      b.String(),
 	}, nil
 }
 
-func convertLogToJSONEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
+func (l *lokiExporter) convertLogToJSONEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
```

Comment:
Unnecessary change?

Reply:
Not exactly, the non-JSON method got changed, I thought it would be better for consistency to have them both part of the lokiExporter.

Reply:
@bogdandrutu, I'm merging this PR as is, but if you think this is a concern, let me know and I'll revert this.